File: Java/AD - HOC/1329 - Cara ou Coroa.java
import java.util.Scanner;

/*
 * Autor: Pierre Vieira
 * Data de submissão: 26/01/2020 14:35:30
 */
public class Main {
    private static int qtdeChar(String plays, char c) {
        int cont = 0;
        for (int i = 0; i < plays.length(); ++i){
            if (plays.charAt(i) == c){
                cont++;
            }
        }
        return cont;
    }

    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        int numero;
        String plays;
        while (true){
            numero = teclado.nextInt();
            if (numero == 0) break;
            teclado.nextLine();
            plays = teclado.nextLine();
            System.out.println("Mary won "+qtdeChar(plays, '0')+" times and John won "+qtdeChar(plays, '1')+" times");
        }
    }
}

File: C/AD - HOC/1329 - Cara ou Coroa.c
/*
 * Autor: Pierre Vieira
 * Data de submissão: 26/01/2020 14:50:48
 */

#include <stdio.h>

int main(){
    int n, numero, qtde_mary, qtde_john;
    while (1){
        qtde_john = qtde_mary = 0;
        scanf("%d", &n);
        if (n == 0){
            break;
        }
        for (int i = 0; i < n; ++i) {
            scanf("%d", &numero);
            if(numero == 0){
                qtde_mary++;
            } else{
                qtde_john++;
            }
        }
        printf("Mary won %d times and John won %d times\n", qtde_mary, qtde_john);
    }
    return 0;
}